plugins/ipam/host-local/backend/allocator/allocator.go

```diff
@@ -203,10 +203,12 @@ func (i *RangeIter) Next() (*net.IPNet, net.IP) {
 
```

Comment:
Good catch! But maybe assigning `i.startIP` before nexting `i.cur` is better than introducing a new variable `nextRange`.

Reply:
Yes, I've thought it, but it will break current logic(some tests failed).

Reply:
Sorry for misread it, do you mean assigning i.startIP before https://github.com/containernetworking/plugins/blob/master/plugins/ipam/host-local/backend/allocator/allocator.go#L205  ?

Reply:
> Sorry for misreading it, do you mean assigning i.startIP before https://github.com/containernetworking/plugins/blob/master/plugins/ipam/host-local/backend/allocator/allocator.go#L205 ?

I have tried it, it will break current tests.

Reply:
Which tests will it break? If the tests rely on internal api, then they can be changed.

Reply:
@squeed Sorry, do you mean assigning i.startIP before nexting i.cur and fix there broken tests in allocator_test.go?

Reply:
> Which tests will it break? If the tests rely on internal api, then they can be changed.

@squeed Sorry, do you mean assigning i.startIP before nexting i.cur and fix there broken tests in allocator_test.go?

Reply:
@mars1024 @squeed If assigning `i.startIP` before nexting `i.cur`, that most recent reserved ip won't be traversed.